### ISSUE TITLE
[DOCKER] Web socket connexion between the bot and web server

### ISSUE THREAD

**Issue by lokiiiii** [PokemonGoF/PokemonGo-Bot#5732]
Hi there,

I've browsed the issues but didn't find one that could help:

It's been a few weeks now that I'm trying to enable the websocket connexion between Web & Bot.

I use Docker with the last "master" commit and I'm using the docker-compose.yml of PokemonGoF/PokemonGo-Bot to start the webserver, the websocket and the bot.

I think that my websocket is well configured but when I access the Web map, I can see my trainer doing its stuff, but the logs won't work (it never did) and since the last commit, I now have an error message saying that it can't access "0.0.0.0:4000" and that it will try again in 30s.

When checking my server, a netstat command shows me that my port 4000 is actually listening right now.

Is there a way to be sure that my websocket is actually listening and doing things ?

My config.json contains: 
- "websocket_server": true,  
and
- "websocket": { "start_embedded_server": true,  "server_url": "0.0.0.0:4000" }  

If someone could help, it would be awesome :)
Thanks all !

**Comment by gise88** [comment]
I don't know if my settings are completely correct but they work:
```
"websocket_server": false,
"websocket": {
	"start_embedded_server": true,
	"server_url": "0.0.0.0:8000"
}
```
and userdata.js inside `web/config/` directory:
```
 var userInfo = {
	users: ["<YOUR-POKEMON-TRAINER-NAME>"],
 	userZoom: false,
 	userPath: true,
 	zoom: 16,
 	userFollow: true,
	gMapsAPIKey: "YourGmapsKey"
};
```

And I have to connect to docker0 ip address (for instance: `172.17.0.1:8000`) or to your ip machine (for instance `192.168.1.10:8000`)

**Comment by lokiiiii** [comment]
@gise88 Thanks a lot for your help.
Which commit of the bot and the web are you using ?
How many containers do you have ? 2 (bot + web) ou 3 (bot + web + socket) ?
The last web server commit adds the websocket adress directly under userdata.js, do you have set it too ?

**Comment by gise88** [comment]
I am using the latest bot dev (commit # https://github.com/PokemonGoF/PokemonGo-Bot/commit/f65b063762384d97c5195a55db1b8257ba6cddee) and the following commit of old web view https://github.com/PokemonGoF/PokemonGo-Web/commit/bd5c2548eadc74de19aca239efefd065b260f4f4.
For what concerns docker I am using docker-compose (so, if i'm not wrong, it spawns 3 containers)!

**Comment by lokiiiii** [comment]
Did you change anything under your docker-compose.yml ?
With the last docker-compose (the one who creates web socket too), by default this web socket runs on port 4000 and docker-compose only expose the port 4000 by default, so I can't understand how you can access this socket by declaring port 8000 under your websocket in config.json 
There must be something that I don't get :p

**Comment by gise88** [comment]
You are right! I have made two huge mistakes.. Sorry! I was at work (as currently I am) and I answered to you relying on my memory! 

First error: in config.json I have `"server_url": "0.0.0.0:4000"`
Second error: I have just retry all my settings and you are correct! 
My web view logs are freezed too.. 
```
10:52

Trainer loaded: MY-TRAINER-NAME
10:52

Data Loaded!
10:52

Loading Data..
```
So, yes! I'm having the same problem..

**Comment by lokiiiii** [comment]
OK so it seems that the actual bot's and websocket's behaviour isn't correct for a Dockerize system ^^
Is anyone able to get websocket work with Docker ?
thanks anyway @gise88 :)

**Comment by mmixx** [comment]
Upon searching the web... I've seen a lot of people having problems connecting docker on their apps using websocket...
